src/temporal.jl

```diff
@@ -0,0 +1,32 @@
+"""
+    HoD <: Transform
+
+Get the hour of day corresponding to the data.
+"""
+struct HoD <: Transform end
+
+
+_apply(x, ::HoD) = hour.(x)
+
+function apply(A::AbstractArray, t::HoD; dims=:, inds=:, kwargs...)
```

Comment:
Do we need `dims` at all for this transform? I think it doesn't matter because it's one-to-one.

Reply:
Ah, looks like it affects returning Matrix vs. vector of vectors. In that case I'm not sure - I don't know if there's a use case for having the result sliced up in different ways. Currently `Periodic` doesn't use`dims`.

Reply:
>I don't know if there's a use case for having the result sliced up in different ways

I don't either, it's a consequence of `eachslice` not that we particularly want it that way.

Reply:
I will note `dims` doesn't make a difference unless `inds` is passed in too for this transform with the updated implementation following this round of code review. 

Also, I'm trying to make this apply method be the canonical apply method that gets used for non-mutating transforms.

Reply:
>Also, I'm trying to make this apply method be the canonical apply method that gets used for non-mutating transforms.

What was the reason this last commit was needed? It seemed out of scope for me but maybe I missed something?

Reply:
> What was the reason this last commit was needed? It seemed out of scope for me but maybe I missed something?

Fair to say it's out of scope. Maybe it should be split up.

Reply:
I wanted to be more clear on the general case, how we want `apply` to work with dims and inds instead of getting lost in the details of hod. Once the general case is clear the individual transform cases become way more trivial

Reply:
fair enough

Reply:
https://github.com/invenia/Transforms.jl/pull/15